JohnBednarczyk/TourTrek
# Feature attraction location directions (#77)

* Update README.md

* Update FriendFragmentTest.java

* Added seraching for attraction

* Added weather to each attraction

* Displaying weather type and fixed tests

* Remove empty test

* Update android-feature.yml

Disable Test Lab

* I add a navigation button which checks for location permission when clicked and then takes the user to a map fragment when permission is given.

* I added a location field to the User class and made a class with Maps utility methods - the first of which is to update the user's last location.

* Pass the context.

* Start the map with the user's location marked.

* A Google Map is displayed which shows the user's last known location, if it is available.

* A marker now shows up indicating the location of the selected attraction, which users can tap to go to Google Maps for navigation.

* update

* I moved the map into the existing AttractionFragment to avoid dealing with issue of MapsFragment not getting attached and maintaining attractionViewModel.

* I removed the MapsFragmentTest.

* A map should not display when adding an attraction.

* Attractions of a tour are now shown on a map. I also fixed the null pointer exception which occurs when the user goes from add tour -> add attraction -> personal tours tab.

* Tests started

* The tour map is now bigger.

* I added a location permission check.

* Improved Efficiency

* Removed firebase test lab

* Delete android-master.yml

* Update android-feature.yml

* Improved read firestore efficiency

* Fixed only retrieving 1 attraction

* There was a reversion to the prior design where the user can tap on a navigation button to get to a new map fragment after correcting the bugs associated with back-pressing and creating view models in the map fragment.

Co-authored-by: John Bednarczyk <johnbednarczykjr@gmail.com>
